experimental/packages/otlp-transformer/src/trace/internal.ts

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import type { Link } from '@opentelemetry/api';
+import { SpanKind, SpanStatusCode } from '@opentelemetry/api';
+import { hrTimeToNanoseconds } from '@opentelemetry/core';
+import type { ReadableSpan, TimedEvent } from '@opentelemetry/sdk-trace-base';
+import { toAttributes } from '../common/internal';
+import { ESpanKind, EStatusCode, IEvent, ILink, ISpan } from './types';
+
+export function toSpan(
+  span: ReadableSpan,
+): ISpan {
+  return {
+    traceId: span.spanContext().traceId,
+    spanId: span.spanContext().spanId,
+    parentSpanId: span.parentSpanId,
+    name: span.name,
+    kind: toKind(span.kind),
+    startTimeUnixNano: hrTimeToNanoseconds(span.startTime),
+    endTimeUnixNano: hrTimeToNanoseconds(span.endTime),
+    attributes: toAttributes(span.attributes),
+    droppedAttributesCount: 0,
+    events: span.events && span.events.map(toSpanEvent),
+    droppedEventsCount: 0,
+    status: {
+      code: toSpanStatusCode(span.status.code),
+      message: span.status.message != null ? span.status.message : undefined,
+    },
+    links: span.links.map(toLink),
+    droppedLinksCount: 0,
+  };
+}
+
+export function toLink(link: Link): ILink {
+  return {
+    attributes: link.attributes ? toAttributes(link.attributes) : [],
+    spanId: link.context.spanId,
+    traceId: link.context.traceId,
+    droppedAttributesCount: 0,
+  };
+}
+
+export function toSpanStatusCode(code: SpanStatusCode): EStatusCode {
+  switch (code) {
+    case SpanStatusCode.OK:
+      return 'STATUS_CODE_OK';
+    case SpanStatusCode.ERROR:
+      return 'STATUS_CODE_ERROR';
+    case SpanStatusCode.UNSET:
+    default:
+      return 'STATUS_CODE_UNSET';
+  }
+}
+
+export function toKind(
+  kind: SpanKind
+): ESpanKind {
+  switch (kind) {
```

Comment:
Lots of potential to improve minified size here. eg. at the top of this file add some global (to this file -- not exported) constants and just reference here (and the STATUS_CODE ones also)

```typescript
const strSpanKind = 'SPAN_KIND_';
const strSpanKindCLIENT = strSpanKind + 'CLIENT';
const strSpanKindCONSUMER = strSpanKind + 'CONSUMER';
const strSpanKindINTERNAL = strSpanKind + 'INTERNAL';
const strSpanKindPRODUCER = strSpanKind + 'PRODUCER';
const strSpanKindSERVER = strSpanKind + 'SERVER';
const strSpanKindUNSPECIFIED = strSpanKind + 'UNSPECIFIED';


// You could also introduce a map here instead of const and just use it, but you would still need "special" code for the undefined
const SpanKindMap = {
    SpanKind.CLIENT: strSpanKind + 'CLIENT',
    ...
};

return SpanKindMap[kind] || strSpanKindUndefined;
```

Reply:
Actually I think @bogdandrutu recommended we use numerics for the enums. The official proto3 mapping suggested strings, but it says in the comments numerics are supported.

Reply:
I changed all these enums to const enums with number values. PTAL